src/App.jsx

```diff
@@ -1,7 +1,48 @@
-export default function App() {
+import { useState } from "react";
+
+const App = () => {
+  const [names, setNames] = useState({ firstName: "", lastName: "" });
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setNames((prevState) => ({ ...prevState, [name]: value }));
+  };
```

Comment:
It is better to use event, but it is also okay to use e. you should change it if you want.
```suggestion
  const handleChange = (event) => {
    const { name, value } = event.target;
    setNames((prevState) => ({ ...prevState, [name]: value }));
  };
```